src/EditorFeatures/CSharpTest2/Recommendations/AssemblyKeywordRecommenderTests.cs

```diff
@@ -116,6 +117,151 @@ public async Task TestInOuterAttribute()
+        public async Task TestBeforeAttributeNamespaceAndAfterUsingWithNoOpenBracket()
+        {
+            await VerifyKeywordAsync(
+@"
+using System;
+
+[$$
+namespace Goo {}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        [WorkItem(362, "https://github.com/dotnet/roslyn/issues/362")]
+        public async Task TestBeforeAssemblyAttributeWithOpenBracket()
+        {
+            await VerifyKeywordAsync(
+@"
+[$$
+[assembly: Whatever]
+namespace Goo {}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        [WorkItem(362, "https://github.com/dotnet/roslyn/issues/362")]
+        public async Task TestAssemblyAttributeBeforeClass()
+        {
+            await VerifyKeywordAsync(
+@"
+[$$
+class Goo {}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        [WorkItem(362, "https://github.com/dotnet/roslyn/issues/362")]
+        public async Task TestAssemblyAttributeBeforeInterface()
+        {
+            await VerifyKeywordAsync(
+@"
+[$$
+interface IGoo {}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        [WorkItem(362, "https://github.com/dotnet/roslyn/issues/362")]
+        public async Task TestAssemblyAttributeBeforeStruct()
+        {
+            await VerifyKeywordAsync(
+@"
+[$$
+struct Goo {}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        [WorkItem(362, "https://github.com/dotnet/roslyn/issues/362")]
+        public async Task TestAssemblyAttributeBeforeEnum()
+        {
+            await VerifyKeywordAsync(
+@"
+[$$
+enum Goo {}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        [WorkItem(362, "https://github.com/dotnet/roslyn/issues/362")]
+        public async Task TestNotBeforeAssemblyAttributeWithoutOpenBracket()
+        {
+            await VerifyAbsenceAsync(
+@"
+$$
+[assembly: Whatever]
+namespace Goo {}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        [WorkItem(362, "https://github.com/dotnet/roslyn/issues/362")]
+        public async Task TestNotBeforeAssemblyAttributeAndAfterUsingWithoutOpenBracket()
+        {
+            await VerifyAbsenceAsync(
+@"
+using System;
+
+$$
+[assembly: Whatever]
+namespace Goo {}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        [WorkItem(362, "https://github.com/dotnet/roslyn/issues/362")]
+        public async Task TestBeforeAssemblyAttributeAndAfterUsingWithoutOpenBracket()
+        {
+            await VerifyKeywordAsync(
+@"
+using System;
+
+[$$
+[assembly: Whatever]
+namespace Goo {}");
+        }
+
         [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
         public async Task TestNotInOuterAttributeInNamespace()
```

Comment:
we need a test that verifies it does not show up inside a namespace.  i.e. ```namespace { [$$ ...``` as well as a test that it doesn't show up after an arbitrary ```[``` i.e. ```class C { void Goo(string[] array) { array[$$ ...```